### PR TITLE
doc: update xline client readme

### PR DESCRIPTION
**Pull request by bsbds**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

   descibed in `Update README` section in #369

* what changes does this pull request make?

    This RP add readme to `xline-client`, and some refactors related to the readme.

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

    Note that some RPs of `xline-client` still not merged. The readme part need to update with these APIs being implemented.

### CODE REVIEW

File: xline-client/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,105 @@
-//! Xline-client
```
Comment (iGxnon) [inline xline-client/src/lib.rs:1]: Could we use `#![doc = include_str!("../README.md")]` here ?
FYI: https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html

File: xline-client/tests/common.rs [hunk 2]
```diff
@@ -4,6 +4,10 @@ use xline_test_utils::Cluster;
 pub async fn get_cluster_client() -> Result<(Cluster, Client)> {
     let mut cluster = Cluster::new(3).await;
     cluster.start().await;
-    let client = Client::connect(cluster.addrs().clone(), ClientOptions::default()).await?;
+    let client = Client::connect(
+        cluster.addrs().iter().collect::<Vec<_>>(),
```
Comment (themanforfree) [inline xline-client/tests/common.rs:8]: Can we only pass the `iterator`` to avoid the heap allocation of `collect`?

File: xline-client/README.md [hunk 3]
```diff
@@ -1,3 +1,112 @@
...
+  - [x] WatchCreate
+  - [x] WatchCancel
+- Auth
+  - [x] Authenticate
+  - [x] RoleAdd
+  - [x] RoleGrantPermission
+  - [x] UserAdd
+  - [x] UserGrantRole
+  - [x] AuthEnable
+  - [x] AuthDisable
```
Comment (themanforfree) [inline xline-client/README.md:32]: Same as above. (`AuthStatus`, `AuthUserDeleteRequest`, etc.)

File: xline-client/src/error.rs [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
 
-/// Xline client result
+/// The result type for `xline-client`
 pub type Result<T> = std::result::Result<T, ClientError>;
 
-/// Client Error
+/// The Error type for `xline-client`
 #[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum ClientError {
```
Comment (Phoenix500526) [inline xline-client/src/error.rs:9]: Some comments on the specific error types are capitalized, like `Rpc Error`,  while the others are lowercase, like `Propose error`. Please use lowercase consistently, say `rpc error`

File: xline-client/src/lib.rs (hunk 1 above)
Comment (bsbds) [reply xline-client/src/lib.rs:1]: done

File: xline-client/README.md [hunk 5]
```diff
@@ -1,3 +1,112 @@
...
+  - [x] Put
+  - [x] Range
+  - [x] Delete
+  - [x] Transaction
+  - [ ] Compact
+- Lease
+  - [ ] Grant
+  - [ ] Revoke
+  - [ ] KeepAlive
+  - [ ] TimeToLive
```
[the remark this answers is not in the document]
Comment (bsbds) [reply xline-client/README.md:21]: done

File: xline-client/tests/common.rs (hunk 2 above)
Comment (bsbds) [reply xline-client/tests/common.rs]: done, now connect accept a IntoIterator

File: xline-client/src/error.rs (hunk 4 above)
Comment (bsbds) [reply xline-client/src/error.rs:9]: done

File: xline-client/src/lib.rs [hunk 6]
```diff
@@ -161,37 +160,41 @@ use utils::config::ClientTimeout;
...
+    /// Kv client
     kv: KvClient,
-    /// lease client
+    /// Lease client
     lease: LeaseClient,
-    /// lock client
+    /// Lock client
     lock: LockClient,
-    /// auth clinet
+    /// Auth clinet
```
Comment (Phoenix500526) [inline xline-client/src/lib.rs:189]: Typos. It's "client", not "clinet"

**Review by Phoenix500526 — COMMENTED** [review]
There is another spelling mistake at line 534 in xline-client/src/types/kv.rs. It's "request", not "reqeust"

File: xline-client/src/lib.rs (hunk 6 above)
Comment (bsbds) [reply xline-client/src/lib.rs]: fixed

**Review by themanforfree — APPROVED** [review]

File: xline-client/src/lib.rs [hunk 7]
```diff
@@ -204,13 +207,32 @@ impl Client {
...
     #[inline]
-    pub async fn connect(
-        all_members: HashMap<String, String>,
-        options: ClientOptions,
-    ) -> Result<Self> {
+    #[allow(clippy::pattern_type_mismatch)] // allow mismatch in map
+    pub async fn connect<E, S>(all_members: S, options: ClientOptions) -> Result<Self>
+    where
+        E: AsRef<str>,
+        S: IntoIterator<Item = (E, E)> + Clone,
```
Comment (iGxnon) [inline xline-client/src/lib.rs:214]: We can remove the Clone trait bound here, so that users can pass in a wider range of possible types.

Comment (bsbds) [reply xline-client/src/lib.rs]: done

**Review by iGxnon — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Review by iGxnon — APPROVED** [review]

**Pull request merged**